File: captchax/sdk/dotnet/src/CaptchaX.SDK/CaptchaXException.cs
namespace CaptchaX.SDK;

public class CaptchaXException : Exception
{
    public int Code { get; }
    public int StatusCode { get; }

    public CaptchaXException(string message, int code = 500, int statusCode = 500)
        : base(message)
    {
        Code = code;
        StatusCode = statusCode;
    }

    public CaptchaXException(string message, Exception innerException)
        : base(message, innerException)
    {
        Code = 500;
        StatusCode = 500;
    }
}

File: captchax/sdk/php/src/CaptchaXException.php
<?php

declare(strict_types=1);

namespace CaptchaX\SDK;

class CaptchaXException extends \Exception
{
    private int $code;
    private int $statusCode;
    private $details;

    public function __construct(
        string $message,
        int $code = 500,
        int $statusCode = 500,
        $details = null,
        ?\Throwable $previous = null
    ) {
        parent::__construct($message, $code, $previous);
        $this->code = $code;
        $this->statusCode = $statusCode;
        $this->details = $details;
    }

    public function getErrorCode(): int
    {
        return $this->code;
    }

    public function getStatusCode(): int
    {
        return $this->statusCode;
    }

    public function getDetails()
    {
        return $this->details;
    }
}
